h/templates/deform/textinput.jinja2

```diff
@@ -1,2 +1,3 @@
 <input type="text"
 {% include "includes/common_attrs.jinja2" %}>
+{%- if field.widget.max_length -%}<span class="form-input__character-counter" data-ref="characterLimitCounter">Up to {{ field.widget.max_length }} characters</span>{% endif %}
```

Comment:
Please avoid over-long lines in templates. They make the markup difficult to read.